test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedDeserializerUndeclaredTests.cs

```diff
@@ -80,12 +80,15 @@ public ODataJsonLightEntryAndFeedDeserializerUndeclaredTests()
         }
         // ----------- end of edm for entry reader ----------- 
 
-        private void ReadEntryPayload(string payload, EdmEntitySet entitySet, EdmEntityType entityType, Action<ODataReader> action, bool readUntypedAsValue = false)
+        private void ReadEntryPayload(string payload, EdmEntitySet entitySet, EdmEntityType entityType, Action<ODataReader> action, bool readUntypedAsValue = false, bool readRequest = false)
         {
             ODataMessageReaderSettings readerSettings = readUntypedAsValue ? UntypedAsValueReaderSettings : UntypedAsStringReaderSettings;
             var message = new InMemoryMessage() { Stream = new MemoryStream(Encoding.UTF8.GetBytes(payload)) };
             message.SetHeader("Content-Type", "application/json");
-            using (var msgReader = new ODataMessageReader((IODataResponseMessage)message, readerSettings, this.serverModel))
+
+            using (var msgReader = readRequest ?
+                new ODataMessageReader((IODataResponseMessage)message, readerSettings, this.serverModel) :
+                new ODataMessageReader((IODataResponseMessage)message, readerSettings, this.serverModel))

```

Comment:
What's the difference between the consequent and the alternative expressions here?

Reply:
+1 -- should line 90 cast message to IODataRequestMessage?

Reply:
Yes.  Thanks.